cpp/FileCompressionType.cpp

```diff
@@ -0,0 +1,199 @@
+  const char * name,
+  bool isSupported){
+  Snowflake::Client::FileCompressionType(fileExtension, name, {}, 0, isSupported);
+}
+
+Snowflake::Client::FileCompressionType::FileCompressionType(
+  const char * fileExtension,
+  const char * name,
+  std::vector<const char *> *magicNumbers,
+  short magicBytes,
+  bool isSupported){
+  m_fileExtension = fileExtension;
+  m_isSupported = isSupported;
+  m_magicNumbers = magicNumbers;
+  m_magicBytes = magicBytes;
+  m_name = name;
+}
+
+const std::vector<const FileCompressionType *> FileCompressionType::types = {
+  &FileCompressionType::GZIP,
+  &FileCompressionType::DEFLATE,
+  &FileCompressionType::RAW_DEFLATE,
+  &FileCompressionType::BZIP2,
+  &FileCompressionType::ZSTD,
+  &FileCompressionType::BROTLI,
+  &FileCompressionType::LZIP,
+  &FileCompressionType::LZMA,
+  &FileCompressionType::LZO,
+  &FileCompressionType::XZ,
+  &FileCompressionType::COMPRESS,
+  &FileCompressionType::ORC,
+  &FileCompressionType::PARQUET,
+};
+
+const FileCompressionType FileCompressionType::GZIP =
+  FileCompressionType(GZIP_EXTENSION, GZIP_NAME, &gzip_magics, 2, true);
+
+const FileCompressionType FileCompressionType::DEFLATE =
+  FileCompressionType(DEFLATE_EXTENSION, DEFLATE_NAME, &deflate_magics, 2, true);
+
+const FileCompressionType FileCompressionType::RAW_DEFLATE=
+  FileCompressionType(RAW_DEFLATE_EXTENSION, RAW_DEFLATE_NAME, true);
+
+const FileCompressionType FileCompressionType::BZIP2 =
+  FileCompressionType(BZIP2_EXTENSION, BZIP2_NAME, &bzip2_magics, 2, true);
+
+const FileCompressionType FileCompressionType::ZSTD =
+  FileCompressionType(ZSTD_EXTENSION, ZSTD_NAME, &zstd_magics, 4, true);
+
+const FileCompressionType FileCompressionType::BROTLI =
+  FileCompressionType(BROTLI_EXTENSION, BROTLI_NAME, &brotli_magics, 4, true);
+
+const FileCompressionType FileCompressionType::ORC =
+  FileCompressionType(ORC_EXTENSION, ORC_NAME, &orc_magics, 3, true);
+
+const FileCompressionType FileCompressionType::PARQUET =
+  FileCompressionType(PARQUET_EXTENSION, PARQUET_NAME, &parquet_magics, 4, true);
+
+const FileCompressionType FileCompressionType::LZIP =
+  FileCompressionType(LZIP_EXTENSION, LZIP_NAME, false);
+
+const FileCompressionType FileCompressionType::LZMA =
+  FileCompressionType(LZMA_EXTENSION, LZMA_NAME, false);
+
+const FileCompressionType FileCompressionType::LZO =
+  FileCompressionType(LZO_EXTENSION, LZO_NAME, false);
+
+const FileCompressionType FileCompressionType::XZ =
+  FileCompressionType(XZ_EXTENSION, XZ_NAME, false);
+
+const FileCompressionType FileCompressionType::COMPRESS =
+  FileCompressionType(COMPRESS_EXTENSION, COMPRESS_NAME, false);
+
+const FileCompressionType FileCompressionType::NONE =
+  FileCompressionType("", "", true);
+
+bool Snowflake::Client::FileCompressionType::matchMagicNumber(char *header) const
+{
+  if (m_magicNumbers)
+  {
+    for (size_t i = 0; i < m_magicNumbers->size(); i++)
+    {
+      if (!memcmp(header, m_magicNumbers->at(i), m_magicBytes))
```

Comment:
You may wan to check if data in header is less than m_magicBytes and return false.